### PR TITLE
feat(core,api-client): Send link previews (WEBFOUND-33)

### PR DESCRIPTION
**Pull request by ffflorian**
## Pull Request Checklist

- [x] My code is covered by tests
- [x] I will [merge the PR as breaking change](https://github.com/wireapp/wire-web-packages/wiki/Releases#create-major-release), if the API contract changes

### CODE REVIEW

File: packages/core/src/demo/echo.js [hunk 1]
```diff
@@ -108,6 +150,7 @@ const messageEchoCache = {};
     const cacheOriginal = assetOriginalCache[messageId];
     if (!cacheOriginal) {
       logger.warn(`Uploaded data for message ID "${messageId} was received before the metadata."`);
```
Comment (atomrc) [inline packages/core/src/demo/echo.js]: I think this is a case we handle in the webapp. If the preview comes before the actual text message, we save it as is. (I believe there was some stories like this in the past with mobile client + bad networking or something).
// ping @gregor

File: packages/core/src/demo/echo.js [hunk 2]
```diff
@@ -91,11 +91,53 @@ const messageEchoCache = {};
...
 
-    await handleIncomingMessage(data);
+      await handleIncomingMessage(data);
+
+      const messageIdOriginal = messageIdCache[messageId];
+
+      if (messageIdOriginal) {
+        textPayload = account.service.conversation.createText(content.text, linkPreviews, messageIdOriginal);
+      } else {
+        logger.warn(`Link preview for message ID "${messageId} was received before the original message."`);
```
Comment (atomrc) [inline packages/core/src/demo/echo.js]: I think this is a case we handle in the webapp. If the preview comes before the actual text message, we save it as is. (I believe there was some stories like this in the past with mobile client + bad networking or something).
// ping @gregor

Comment (atomrc) [reply packages/core/src/demo/echo.js]: Ok nevermind, this is just demo code.

File: packages/api-client/src/test/node/spec/asset/AssetAPI.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+const {APIClient} = require('@wireapp/api-client');
+
+describe('"AssetAPI"', () => {
```
Comment (atomrc) [inline packages/api-client/src/test/node/spec/asset/AssetAPI.test.js]: :+1:

Comment (ffflorian) [reply packages/api-client/src/test/node/spec/asset/AssetAPI.test.js]: :smirk:

**Pull request merged**